src/vivarium_cluster_tools/runner.py

```diff
@@ -237,27 +237,27 @@ def build_job_list(ctx):
     return jobs
 
 
-def np_concat_preserve_types(df_list):
+def concat_preserve_types(df_list):
     dtypes = df_list[0].dtypes
+    dtypes = [list(d[1].index) for d in dtypes.groupby(dtypes)]
```

Comment:
Don't you just need `list(dtypes.unique())` here?

Reply:
no because I'm grouping by dtypes so I end up with a list of lists where each list contains the columns for a certain dtype

Reply:
Oh, nevermind.

Use unpacking though:  ``for _, dtype_group in dtypes.groupby(dtypes)``

Reply:
but I named that very confusingly. I'll update to a more self-explanatory name

Reply:
just updated to have a clearer name + use unpacking